systemtest/processor_test.go

```diff
@@ -71,46 +70,48 @@ func TestHotStandby(t *testing.T) {
 		goka.WithStorageBuilder(proc2Storages.Build),
 	)
 
-	test.AssertNil(t, err)
+	require.NoError(t, err)
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	errg, ctx := multierr.NewErrGroup(ctx)
 
 	errg.Go(func() error {
-		return proc1.Run(ctx)
+		// simulatenously stopping multiple processors sometimes fails the processors, so this one gets delayed
+		// see issue #376 for details
+		return proc1.Run(DelayedCtxCloser(ctx, 5*time.Second))
```

Comment:
workaround to make the test more stable: avoid shutting down the processors simultaneously